libnfc/nfc.c

```diff
@@ -579,6 +579,7 @@ nfc_initiator_select_passive_target(nfc_device *pnd,
     free(abtTmpInit);
   } else if (nm.nmt == NMT_ISO14443A) {
     abtInit = abtTmpInit;
+    free(abtTmpInit);
     iso14443_cascade_uid(pbtInitData, szInitData, abtInit, &szInit);
```

Comment:
Here is now a use-after-free for abtInit (which is assigned the address of abtTmpInit which is freeed).

Reply:
Agree with @smortex , this fix doesn't make sense.  abtInit pointing to the freed memory location, and is subsequently passed to iso14443_cascade_uid.